File: src/main/java/io/openfuture/openmessanger/repository/entity/PrivateChat.java
package io.openfuture.openmessanger.repository.entity;


import java.util.List;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor

@Entity
@Table(name = "private_chat")
public class PrivateChat {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(name = "type")
    private String type;

    @OneToMany(mappedBy = "chatId")
    private List<ChatParticipant> chatParticipants;

    public PrivateChat(final String type) {
        this.type = type;
    }

}

File: src/main/kotlin/io/openfuture/openmessanger/repository/entity/PrivateChat.kt
package io.openfuture.openmessanger.repository.entity

import jakarta.persistence.*

@Entity
@Table(name = "private_chat")
class PrivateChat(
    @Column(name = "type")
    val type: String
) {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Int? = null

    @OneToMany(mappedBy = "chatId")
    var chatParticipants: List<ChatParticipant>? = null
}